4_predict/src/train_models.R

```diff
@@ -555,6 +555,228 @@ filter_rows_date <- function(attrs, start_date){
 
 
 #SHAP values
+setup_shap_data <- function(data, seasonal = TRUE, physio = TRUE, lulc = TRUE,
+                            lulc_prop = 0.75, lulc_data = NULL, physio_data = NULL){
+  #' @description prepares data for SHAP computation based on the provided split
+  #'
+  #' @param data the dataset with predictor attributes
+  #' @param seasonal logical for splitting data by water year season
```

Comment:
I found these argument descriptions helpful, thanks! The arguments themselves might be more intuitive by rewording slightly, e.g. `split_by_season` instead of `seasonal`. This is a very minor suggestion and doesn't influence the outcome of the code, so I leave this up to you.